commit fb3aa34cb2a90427f8059990eb505c11aed58d84
Author: Daniel P. Berrange <berrange@redhat.com>
Date:   Wed Feb 15 14:06:55 2012 +0000

    Fix bandwidth units for block_pull/block_rebase

diff --git a/lib/Sys/Virt/Domain.pm b/lib/Sys/Virt/Domain.pm
index 5e98b62..98daa78 100644
--- a/lib/Sys/Virt/Domain.pm
+++ b/lib/Sys/Virt/Domain.pm
@@ -907,13 +907,13 @@ associated with C<$path>
 
 Merge the backing files associated with C<$path> into the
 top level file. The C<$bandwidth> parameter specifies the
-maximum I/O rate to allow in KB/s.
+maximum I/O rate to allow in MB/s.
 
 =item $dom->block_rebase($path, $backingpath, $bandwith, $flags=0)
 
 Switch the backing path associated with C<$path> to instead
 use C<$backingpath>. The C<$bandwidth> parameter specifies the
-maximum I/O rate to allow in KB/s.
+maximum I/O rate to allow in MB/s.
 
 =item $count = $dom->num_of_snapshots()
 
commit 08624860ea65b27edaf4cf20387e935d5c242482
Author: Daniel P. Berrange <berrange@redhat.com>
Date:   Wed Feb 15 14:09:56 2012 +0000

    Fix bandwidth units for block_job_set_speed

diff --git a/lib/Sys/Virt/Domain.pm b/lib/Sys/Virt/Domain.pm
index 98daa78..78da71e 100644
--- a/lib/Sys/Virt/Domain.pm
+++ b/lib/Sys/Virt/Domain.pm
@@ -896,7 +896,7 @@ qualified path of the block device being changed.
 
 Change the maximum I/O bandwidth used by the block job that
 is currently executing for C<$path>. The C<$bandwidth> argument
-is specified in KB/s
+is specified in MB/s
 
 =item $dom->abort_block_job($path, $flags=0)
 
