commit d8f1122f47b7d7b6528aceed31152f43e8897f7d
Author: Daniel P. Berrange <berrange@redhat.com>
Date:   Wed Apr 4 10:35:29 2012 +0100

    Add an example of error handling to docs

diff --git a/lib/Sys/Virt.pm b/lib/Sys/Virt.pm
index d036961..24b5294 100644
--- a/lib/Sys/Virt.pm
+++ b/lib/Sys/Virt.pm
@@ -45,7 +45,14 @@ a consistent API.
 Any operations in the Sys::Virt API which have failure scenarios
 will result in an instance of the L<Sys::Virt::Error> module being
 thrown. To catch these errors, simply wrap the method in an eval
-block. For details of the information contained in the error objects,
+block:
+
+  eval { my $vmm = Sys::Virt->new(address => $addr); };
+  if ($@) {
+    print STDERR "Unable to open connection to $addr" . $@->message . "\n";
+  }
+
+For details of the information contained in the error objects,
 consult the L<Sys::Virt::Error> manual page.
 
 =head1 METHODS
