commit 6608ce471c40589563e1811d445466facd290897
Author: Daniel P. Berrange <berrange@redhat.com>
Date:   Fri Mar 16 11:28:11 2012 +0000

    Fix units for migration bandwidth docs

diff --git a/lib/Sys/Virt/Domain.pm b/lib/Sys/Virt/Domain.pm
index 78da71e..b55a204 100644
--- a/lib/Sys/Virt/Domain.pm
+++ b/lib/Sys/Virt/Domain.pm
@@ -812,13 +812,13 @@ The C<$flags> parameter is currently unused and defaults to zero.
 =item $dom->migrate_set_max_speed($bandwidth, $flags)
 
 Set the maximum allowed bandwidth during migration of the guest.
-The C<bandwidth> parameter is measured in kilobytes/second.
+The C<bandwidth> parameter is measured in MB/second.
 The C<$flags> parameter is currently unused and defaults to zero.
 
 =item $bandwidth = $dom->migrate_get_max_speed($flag)
 
 Get the maximum allowed bandwidth during migration fo the guest.
-The returned <bandwidth> value is measured in kilobytes/second.
+The returned <bandwidth> value is measured in MB/second.
 The C<$flags> parameter is currently unused and defaults to zero.
 
 =item $dom->inject_nmi($flags)
