commit 9aab5b096e035064f328c74a55d0a1cf0d1dc29f
Author: Daniel P. Berrange <berrange@redhat.com>
Date:   Wed Aug 3 17:21:25 2011 +0100

    Re-add virDomainAbortJob binding accidentally removed

diff --git a/Virt.xs b/Virt.xs
index e798a27..b8e17f9 100644
--- a/Virt.xs
+++ b/Virt.xs
@@ -2258,6 +2258,15 @@ get_job_info(dom)
 
 
 void
+abort_job(dom)
+      virDomainPtr dom;
+    PPCODE:
+      if (virDomainAbortJob(dom) < 0) {
+	_croak_error(virGetLastError());
+      }
+
+
+void
 abort_block_job(dom, path, flags=0)
       virDomainPtr dom;
       const char *path;
